SoftLayer/managers/dedicated_host.py

```diff
@@ -33,10 +33,138 @@ def __init__(self, client, ordering_manager=None):
         self.client = client
         self.account = client['Account']
         self.host = client['Virtual_DedicatedHost']
+        self.guest = client['Virtual_Guest']
 
         if ordering_manager is None:
             self.ordering_manager = ordering.OrderingManager(client)
 
+    def cancel_host(self, host_id):
+        """Cancel a dedicated host immediately, it fails if there are still guests in the host.
+
+        :param host_id: The ID of the dedicated host to be cancelled.
+        :return: True on success or an exception
+
+        Example::
+            # Cancels dedicated host id 12345
+            result = mgr.cancel_host(12345)
+
+        """
+        return self.host.deleteObject(id=host_id)
+
+    def cancel_guests(self, host_id):
+        """Cancel all guests into the dedicated host immediately.
+
+        To cancel an specified guest use the method VSManager.cancel_instance()
+
+        :param host_id: The ID of the dedicated host.
+        :return: True on success, False if there isn't any guest or
+                 an exception from the API
+
+        Example::
+            # Cancel guests of dedicated host id 12345
+            result = mgr.cancel_guests(12345)
+        """
+        result = False
+
+        guests = self.host.getGuests(id=host_id, mask='id')
+
+        if guests:
+            for vs in guests:
+                result = self.guest.deleteObject(id=vs['id'])
```

Comment:
Need to wrap this in a try/catch for SoftLayer.Exceptions. I can see a case where one guest will have a running transaction and won't be able to be canceled. So we don't want this command to fail midway.

Reply:
Done, is in the method called `_delete_guest`. I also added a unit-test to verify the exception message is retrieved in case the deleteObject fails